### ISSUE TITLE
Get integer from inside an optional block

### ISSUE THREAD

**Issue by DarkShadow44** [kaitai-io/kaitai_struct#502]
Now, the problem is a bit hard to describe:
The file contains list of blocks, which are identified by an id in front of each block. One of the blocks contains a number I need later on in the parsing, but outside the block. Since I don't know which index the block has, or if it even exists, I can't get that integer.
See the code for a clearer view of my problem:
```yaml
meta:
  id: ribbon
  endian: le

seq:
  - id: count_blocks
    type: u1
  - id: blocks
    type: type_block_generic
    repeat: expr
    repeat-expr: count_blocks
  - id: main_blocks
    type: u2
    repeat: expr
    repeat-expr: count_optional_tabs # <--problem

enums:
  block_type:
    0: main_tabs
    1: optional_tabs

types:
  type_block_main_tabs:
    seq:
    - id: count_main_tabs
      type: u2
  
  type_block_optional_tabs:
    seq:
    - id: count_optional_tabs
      type: u2
      
  type_block_generic:
    seq:
    - id: type
      type: u2
      enum: block_type
    - id: block
      type:
        switch-on: type
        cases:
          block_type::main_tabs: type_block_main_tabs
          block_type::optional_tabs: type_block_optional_tabs
```
count_optional_tabs needs to be used later, and it can't be accessed outside. And if it doesn't exist, I need to skip the field (i.e. assume count_optional_tabs as 0).
Do we already have functionality for that kind of issue? Sorry if I missed something, but I didn't find anything.

**Comment by KOLANICH** [comment]
1. You can't do it in pure KS.
2. But please look into real files and check if the blocks with type_block_optional_tabs contain anything except count. Look, there may be more than one of such blocks. What will the lib do in this case? Will it read them sequentially from the end of stream? Will it crash? Will it refuse to parse file? Is it possible that `main_blocks` is just a yet another block (it is very strange and suspicious that it is not, could you check it)? Does the lib assume the rigid sequence of block types (like [CNTK does](https://github.com/KOLANICH/kaitai_struct_formats/blob/CNTK/scientific/data_science/cntk.ksy)) and can it be exploited?

If nothing helps, you can either:
1 call an opaque function from ksy retrieving the content of the needed block.
2 generate the code, hand-patch it, generate a patch, so you can automatically apply it to the code generated from ksy.

**Comment by DarkShadow44** [comment]
Hi and thanks for the quick response!

I fixed the syntax highlighting. Not sure what you mean about the indentation though, all my files look like this.

a) I already figured out that the order of blocks is not fixed, they can be rearranged and it still needs to work.
b) Every unique block type can only occur once
c)  'main_blocks' is not part of 'blocks', I got a size and a count for the blocks - it doesn't fit in there

Also, keep in mind that this a simplified example, it's a bit more complicated in the actual files, but I figured you wouldn't want the hundreds of lines it takes to parse the real files.

Regarding offsets I need to check, could be possible to parse it like this.
